peer/test_utils.go

```diff
@@ -726,3 +727,79 @@ func createTestPeer(t *testing.T) *testPeerParams {
 		chanStatusMgr: chanStatusMgr,
 	}
 }
+
+func startPeer(t *testing.T, mockConn *mockMessageConn,
+	peer *Brontide, extraChecks ...func() error) {
```

Comment:
Please don't do this `extraChecks` thing. It doesn't operate on any values that are only accessible from the inner scope and makes the API much worse.

Reply:
What are your suggestions on extending this function. See how it is used her:
https://github.com/lightningnetwork/lnd/blob/15c31070fa874a5036aec01561661e9054035bae/peer/brontide_test.go#L1465-L1495

Reply:
just run that function after `startPeer` there's no reason for it to be inside of the `startPeer` scope.

Reply:
The function takes no arguments so its entire closure environment is captured prior to the run. If you are depending on the right ordering of state mutations of values defined outside the environment then this design is bad as it is thoroughly unintuitive and offers no explanation to the caller what should or should not be supplied in this list. Find another way.

Reply:
If I make the closure accept `mockconn` as an argument, would it make this okay?

Reply:
I removed it, I will just add a bool value to it  and write the test in the goroutine itself, when I need to extend,